### DISCUSSION TITLE
Suppress toast stdout output?

### DISCUSSION THREAD

**Discussion by InterStella0** [GitHub30/win11toast#4] [Q&A] [answered]
Using the example 
```python
from win11toast import toast

toast('Hello Pythonista', 'Click to run python script', on_click=r'C:\Users\Admin\Downloads\handler.pyw')
```
Outputs
```
ToastDismissalReason.USER_CANCELED
```
Is it intentional for it to output that? If so, is there a way to suppress it? Right now I'm using `contextlib.redirect_stdout` but it would be a little bit convenient to have something to turn that off.

**Comment by GitHub30** [comment, marked as answer]
Thanks for your question. You can override the handler for on_dismissed.

```python
from win11toast import toast

def empty_func(args):
    pass

toast('Hello Pythonista', 'Click to run python script', on_click=r'C:\Users\Admin\Downloads\handler.pyw', on_dismissed=empty_func)
```

https://user-images.githubusercontent.com/12811398/188865923-fe42dd13-8f18-4aef-808f-0dd5b0f7b4cb.mp4

https://github.com/GitHub30/win11toast/blob/38ff6575b036cacee6cfd15c30a2f43ce2b7c1a8/win11toast.py#L298

I'm not familiar with asyncio, but I would like to return the result as the return value of toast instead of standard output.
https://github.com/GitHub30/win11toast/discussions/3#discussioncomment-3526753

**Reply by GitHub30** [reply]
shorthand
https://stackoverflow.com/questions/22738412/a-suitable-do-nothing-lambda-expression-in-python

```python
from win11toast import toast

toast('Hello Pythonista', 'Click to run python script', on_click=r'C:\Users\Admin\Downloads\handler.pyw', on_dismissed=lambda *args: None)
```

**Reply by InterStella0** [reply]
Understood, I just realized that most of the on_* params callbacks are set to print. Thank you
